### DISCUSSION TITLE
Why does contact storage need to be enabled to show call history?

### DISCUSSION THREAD

**Discussion by jekv2** [FossifyOrg/General-Discussion#527] [Q&A]
Why does contacts storage needs to be enabled to show call history on phone app? If contacts storage is disabled, nothing displays in phone call history.

ty

![Screenshot_20241126-112018](https://github.com/user-attachments/assets/5fd1b285-1204-4ad1-ba08-3dab50a2f647)

**Comment by Aga-C** [comment]
In the current [implementation](https://github.com/FossifyOrg/Phone/blob/0a6021a294573155540bbd20b6eb995c19336ec1/app/src/main/kotlin/org/fossify/phone/helpers/RecentsHelper.kt#L120) creating call history relies on the access to contacts to fetch the data. If it can't fetch them, it just returns the empty list.
